pkg/apis/core/v1alpha1/workflow_methods.go

```diff
@@ -30,3 +30,83 @@ func (w *Workflow) GetStartTime() *metav1.Time {
 	}
 	return nil
 }
+
+type taskInfo struct {
+	CurrentWorker        string
+	CurrentTask          string
+	CurrentTaskIndex     int
+	CurrentAction        string
+	CurrentActionIndex   int
+	CurrentActionState   string
+	TotalNumberOfActions int
+}
+
+// helper function for task info.
+func (w *Workflow) getTaskActionInfo() taskInfo {
+	var (
+		found           bool
+		taskIndex       = -1
+		actionIndex     int
+		actionTaskIndex int
+		actionCount     int
+	)
+	for ti, task := range w.Status.Tasks {
+		for ai, action := range task.Actions {
+			actionCount++
+			// Find either the first action in a non-terminal state or the first action in an unsuccessful terminal state
+			if (action.Status == "STATE_PENDING" || action.Status == "STATE_RUNNING" ||
+				action.Status == "STATE_FAILED" || action.Status == "STATE_TIMEOUT") && !found {
```

Comment:
There should likely be constants for these statuses here.

Reply:
Sure, I can add these